ct-app/core/node.py

```
@@ -45,19 +45,27 @@
+
         self.params = Parameters()
-        self.message_distributed = dict[str, int]()
+        self.messages_distributed = dict[str, int]()
 
+        self.connected = False
         self.running = False
 
+    @property
+    async def safe_address(self):
+        if self._safe_address is None:
+            if info := await self.api.node_info():
```

Ruff

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)